src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,132 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                book.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot insert book into DB", e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();;
+            if (resultSet.next()) {
+                Long bookId = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+
+                Book book = new Book();
+                book.setId(bookId);
+                book.setTitle(title);
+                book.setPrice(price);
```

Comment:
When converting `ResultSet` to `Book`, choose either to use setters or constructor for object initialization, but not both. It's inconsistent to use both ways.